tracetools/src/tracetools.c

```diff
@@ -489,6 +491,17 @@ DEFINE_TRACEPOINT(
   TRACEPOINT_ARGS(
     buffer))
 
+void __attribute__((constructor)) tracetools_init()
+{
+  if(getenv("TRACETOOLS_DISABLE")) {
```

Comment:
I would give it a different name just to avoid any confusion with the existing `TRACETOOLS_DISABLED` CMake option/define. Maybe `TRACETOOLS_RUNTIME_DISABLE`?

```suggestion
  if(getenv("TRACETOOLS_RUNTIME_DISABLE")) {
```

The environment variable should also be documented. I wouldn't put it [under the "Building" section next to the documentation about `TRACETOOLS_DISABLED`/`TRACETOOLS_TRACEPOINTS_EXCLUDE`](https://github.com/ros2/ros2_tracing#building) since this isn't a build thing. Maybe as a new section after the ["Tracing" section](https://github.com/ros2/ros2_tracing#tracing)? For example, an "Opt-out from tracing at runtime" section where this mechanism and the env var are explained.

Reply:
Done in 814bdd421e14cb4a6a469719edbe1e39f96fab99